src/probability/probability_mass_function/basic.lean

```diff
@@ -239,11 +238,60 @@ lemma to_measure_apply_fintype [fintype α] : p.to_measure s = ∑ x, s.indicato
 
 end measurable_singleton_class
 
+end measure
+
+end pmf
+
+namespace measure_theory
+
+open pmf
+
+namespace measure
+
+/-- Given that `α` is a countable, measurable space with all singleton sets measurable,
+we can convert any probability measure into a `pmf`, where the mass of a point
+is the measure of the singleton set under the original measure. -/
+def to_pmf [countable α] [measurable_space α] [measurable_singleton_class α]
+  (μ : measure α) [h : is_probability_measure μ] : pmf α :=
+⟨λ x, μ ({x} : set α), ennreal.summable.has_sum_iff.2 (trans (symm $
+(tsum_indicator_apply_singleton μ set.univ measurable_set.univ).symm.trans
+  (tsum_congr (λ x, congr_fun (set.indicator_univ _) x))) (h.measure_univ))⟩
+
+variables [countable α] [measurable_space α] [measurable_singleton_class α]
+  (μ : measure α) [is_probability_measure μ]
+
+lemma to_pmf_apply (x : α) : μ.to_pmf x = μ {x} := rfl
+
+@[simp] lemma to_pmf_to_measure : μ.to_pmf.to_measure = μ :=
+measure.ext (λ s hs, by simpa only [μ.to_pmf.to_measure_apply s hs,
+  ← μ.tsum_indicator_apply_singleton s hs])
+
+end measure
+
+end measure_theory
+
+namespace pmf
+
+open measure_theory
+
 /-- The measure associated to a `pmf` by `to_measure` is a probability measure -/
-instance to_measure.is_probability_measure (p : pmf α) : is_probability_measure (p.to_measure) :=
+instance to_measure.is_probability_measure [measurable_space α] (p : pmf α) :
+  is_probability_measure (p.to_measure) :=
 ⟨by simpa only [measurable_set.univ, to_measure_apply_eq_to_outer_measure_apply,
   set.indicator_univ, to_outer_measure_apply, ennreal.coe_eq_one] using tsum_coe p⟩
 
-end measure
+variables [countable α] [measurable_space α] [measurable_singleton_class α]
+  (p : pmf α) (μ : measure α) [is_probability_measure μ]
+
+lemma to_measure_to_pmf : p.to_measure.to_pmf = p :=
+pmf.ext (λ x, by rw [← p.to_measure_apply_singleton x (measurable_set_singleton x),
+  p.to_measure.to_pmf_apply])
+
+lemma to_measure_eq_iff_to_pmf_eq (μ : measure α) [hμ : is_probability_measure μ]:
+  p.to_measure = μ ↔ μ.to_pmf = p :=
+begin
+  refine ⟨λ h, trans _ p.to_measure_to_pmf, λ h, trans (congr_arg _ h.symm) μ.to_pmf_to_measure⟩,
+  unfreezingI {induction h, refl}
+end
```

Comment:
```suggestion
lemma to_measure_injective : injective (to_measure : pmf α → measure α) := sorry

@[simp] lemma to_measure_inj {p q : pmf α} : p.to_measure = q.to_measure ↔ p = q :=
to_measure_injective.eq_iff

lemma to_measure_eq_iff_to_pmf_eq {p : pmf α} {μ : measure α} [hμ : is_probability_measure μ] :
  p.to_measure = μ ↔ p = μ.to_pmf :=
by rw [←to_measure_inj, to_measure_to_pmf]
```

Reply:
I added these injectivity lemmas a bit earlier since they don't require that the space is countable, unlike these lemmas involving `to_pmf`. Also added versions for `to_outer_measure` and not just `to_measure`

Reply:
You don't seem to have pushed your changes.

Reply:
Sorry about that, I had pushed to a sub-branch instead of this